src/components/Heading/Heading.tsx

```diff
@@ -13,40 +27,53 @@ type Props = {
    * and the `size` will be used to determine the styling.
    */
   as?: HeadingElement;
-  children: TypographyProps<HeadingElement>["children"];
-  className?: TypographyProps<HeadingElement>["className"];
-  color?: TypographyProps<HeadingElement>["color"];
-  size: TypographyProps<HeadingElement>["size"];
-  spacing?: TypographyProps<HeadingElement>["spacing"];
+  children: React.ReactNode;
+  className?: string;
+  size: HeadingElement;
   tabIndex?: number;
-  weight?: TypographyProps<HeadingElement>["weight"];
+  variant?: Variant;
 } & React.HTMLAttributes<HTMLHeadingElement>;
 
 /**
  * ```ts
- * import {Heading} from "@chanzuckerberg/eds";
+ * import {Heading} from "@chanzuckerberg/eds-components";
  * ```
  */
 
-const Heading = forwardRef<HTMLElement, Props>(
+export const Heading = forwardRef(
   (
     {
       as,
       children,
+      className,
+      variant,
       size,
       /**
        * Components that wrap typography sometimes require props such as
-       * event handlers, tabIndex, etc. to be passed down into the element.
-       * TODO: add better typing or documentation for optional props,
-       * e.g. React.HTMLAttributes<HTMLHeadingElement>
-       */ ...rest
+       * event handlers, tabIndex, etc. and/or other native heading element
+       * attributes to be passed down into the element.
+       */ ...other
     }: Props,
-    ref,
-  ) => (
-    <Typography as={as || size} ref={ref} size={size} {...rest}>
-      {children}
-    </Typography>
-  ),
+    ref: React.ForwardedRef<HTMLHeadingElement>,
+  ) => {
+    if (variant === "info" && process.env.NODE_ENV !== "production") {
+      console.warn(
+        "Info variant is deprecated and will be removed in an upcoming release. Please use the consider another variant instead.",
```

Comment:
Maybe we should add test for these deprecation warnings.

Reply:
Sounds like a good idea, will do